### PR TITLE
feat: implement presentation activity view (#93)

### LINKED ISSUE #93: Implement Presentation Activity View
### Description
Allow users to review previously completed presentation transactions for audit and transparency.

### Tasks
- [ ] Create `PresentationActivityPage` component
- [ ] Create `PresentationActivityItem` component
- [ ] Call backend endpoint `GET /presentation/activity` to fetch records
- [ ] Display presentation records in chronological order
- [ ] Display verifier information (name, logo, client_id)
- [ ] Display credential types shared
- [ ] Display claim summary (count, not full values for privacy)
- [ ] Display presentation timestamp
- [ ] Add details view for individual presentations
- [ ] Add pagination or infinite scroll for large datasets
- [ ] Add filter by date range
- [ ] Add search by verifier name
- [ ] Add delete/forget option for privacy

### Privacy Considerations
- [ ] Allow users to delete history entries
- [ ] Auto-expire history after configurable period (default 30 days)
- [ ] Don't store sensitive claim values, only metadata
- [ ] Confirm before deleting

### Acceptance Criteria
- User can see past presentations in chronological order
- Presentation details are accessible via details view
- Records ordered by date (newest first)
- UI remains performant with large datasets (pagination)
- User can delete entries for privacy
- Auto-expiry of old records

### Backend Dependencies
- [ ] `GET /presentation/activity` - Return presentation history records
- [ ] `DELETE /presentation/activity/:id` - Delete specific record

### PR DESCRIPTION
**Pull request by Vitalisn4**
### Summary

Implements Presentation Activity (#93): Activity History list, Shared Claims detail, and delete confirmation — aligned with Figma and wired to the live API.

### Changes

- **Activity History** (`/activity`) — sticky header/footer, scrollable cards, settings icon
- **Shared Claims** (`/activity/:id`) — full-width detail view from card click
- **Cards** — full-card click, mint hover (`#e6f4e6`), trash opens delete modal
- **Delete modal** — Figma copy; red **Yes, delete activity** button
- **API** — `GET/DELETE /presentation/activity`; mock data and filter UI removed

## Test plan

- [ ] Activity list loads from backend (or shows empty state)
- [ ] Header/footer stay fixed while scrolling
- [ ] Card click opens Shared Claims; hover shows mint green
- [ ] Delete flow works via API; Cancel closes modal
- [ ] `npm test` passes for presentation activity modules

Closes #93

### CODE REVIEW

**Comment by Ngha-Boris** [comment]
Hello @Vitalisn4 
Please dispay a better error here 
image

**Comment by Ngha-Boris** [comment]
> Hello @Vitalisn4 Please dispay a better error here image

I don't even think we should have an error massage like the one in red on this page, @mbunwe-victor what do you think?

**Comment by mbunwe-victor** [comment]
> > Hello @Vitalisn4 Please dispay a better error here image
> 
> I don't even think we should have an error massage like the one in red on this page, @mbunwe-victor what do you think?

Yes, we shouldn't have the error displayed on this page, after which we get the note saying there was no "past activity," which I also understand because we are calling the backend to check for the various activities, which is not yet fully implemented.

We should gladly just move to the Activity page for now, without the error pop-up.

File: .env.example [hunk 1]
```diff
@@ -17,4 +17,6 @@ VITE_API_BASE_URL=http://localhost:3000
 # Example:
 # VITE_ALLOWED_CREDENTIAL_OFFER_HOSTS=issuer.example.com,wallet.example.org,localhost:8080
 # VITE_ALLOWED_CREDENTIAL_OFFER_HOSTS=
-
+#
+# Presentation activity history retention in days (default 30 when unset or invalid).
+# VITE_PRESENTATION_ACTIVITY_RETENTION_DAYS=30
```
Comment (mbunwe-victor) [inline .env.example]: Is this comment really necessary for now?

**Review by mbunwe-victor — COMMENTED** [review]
Hello @Vitalisn4, thank you so much for this PR. I have a concern.
So, going through the implementation, I noticed you added a certain page, `PresentationActivityDetailsPage,` and I don't recall us having that, or maybe it's a missing design. Does it follow the happy path?

**Review by Ngha-Boris — CHANGES_REQUESTED** [review]
## Code Review: Presentation Activity View Implementation

### Summary
The implementation largely meets the ticket requirements with good overall code quality, strong type safety, and proper privacy considerations. Most acceptance criteria are satisfied including chronological ordering, pagination, filtering, deletion with confirmation, and privacy safeguards.

### ✅ Requirements Met
- `PresentationActivityPage` and `PresentationActivityItem` components created
- `GET /presentation/activity` and `DELETE /presentation/activity/:id` endpoints integrated
- Chronological ordering (newest first) with proper sorting
- Verifier information (name, logo, client_id) displayed
- Credential types and claim count (metadata only) shown
- Details view with `PresentationActivityDetailPage`
- Pagination with "Load more" pattern
- Date range and verifier name filters
- Delete/forget option with confirmation dialog
- Auto-expiry configuration (30 days default)
- No sensitive claim values stored

---

### 🔴 Issues Requiring Attention

#### Issue 1: Race Condition in Filter Debounce
**File:** `src/hooks/usePresentationActivity.ts` (lines 68-71)

The debounce pattern can cause stale filter updates to overwrite newer ones if filters change rapidly within the 350ms window.

```typescript
useEffect(() => {
  const timeoutId = window.setTimeout(() => setAppliedFilters(filters), 350)
  return () => window.clearTimeout(timeoutId)
}, [filters])
```

**Suggestion:** Use a ref to track latest filters:
```typescript
const latestFilters = useRef(filters)
latestFilters.current = filters

useEffect(() => {
  const timeoutId = window.setTimeout(() => {
    setAppliedFilters(latestFilters.current)
  }, 350)
  return () => window.clearTimeout(timeoutId)
}, [filters])
```

---

#### Issue 2: Stale Closure in removeItem Callback
**File:** `src/hooks/usePresentationActivity.ts` (lines 122-130)

The `setHasMore` calculation uses `page` and `total` from closure which may be stale due to React's batched state updates.

```typescript
setHasMore(page * DEFAULT_PRESENTATION_ACTIVITY_PAGE_SIZE < total - 1)
```

**Suggestion:** Calculate reactively in a separate `useEffect` based on `items.length`, `page`, and `total`, or use functional updates to ensure consistency.

---

#### Issue 3: Error State Not Cleared on Successful Delete
**File:** `src/hooks/usePresentationActivity.ts` (lines 122-130)

After successful deletion, any previous error message remains displayed.

**Suggestion:** Add `setErrorMessage(null)` or call `clearError()` after successful deletion in the `removeItem` function.

---

### 🟡 Suggestions for Improvement

1. **Date Range Validation:** Add validation to ensure `from` date is not after `to` date in `PresentationActivityFilters.tsx`

2. **Accessibility Enhancement:** Add `aria-busy={loadingMore}` to the "Load more" button for better screen reader support

3. **Zero Claims Message:** Consider showing "No claims shared" instead of "0 claims shared" for better UX

4. **Test Coverage:** Add tests for:
   - `PresentationActivityDetailPage` error states
   - `DeletePresentationActivityDialog` interactions
   - Filter functionality and edge cases

---

### Positive Highlights

- ✅ Strong privacy practices (metadata only, no claim values)
- ✅ Defensive client-side retention filtering
- ✅ Proper UUID validation before API calls
- ✅ Context-specific, user-friendly error messages
- ✅ Good accessibility with ARIA labels and roles
- ✅ Comprehensive TypeScript types with runtime validation
- ✅ Proper AbortController cleanup for API requests
- ✅ Clean architecture with separation of concerns

---

**Overall:** Good implementation that meets the requirements. Please address the race condition (Issue 1) and stale closure (Issue 2) before merging, as these could cause subtle bugs in production.

**Comment by Vitalisn4** [comment]
Here are the different screens 
image
image
image

File: src/components/Header.tsx
Comment (mbunwe-victor) [inline src/components/Header.tsx:1]: I honestly don't know why we have all these changes here

File: src/pages/PresentationActivityPage.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                    strokeLinecap="round"
+                    strokeLinejoin="round"
+                    d="M12 15.5a3.5 3.5 0 1 0 0-7 3.5 3.5 0 0 0 0 7Z"
+                  />
+                  <path
+                    strokeLinecap="round"
+                    strokeLinejoin="round"
+                    d="M19.4 15a1.7 1.7 0 0 0 .3 1.9l.1.1a2 2 0 1 1-2.8 2.8l-.1-.1a1.7 1.7 0 0 0-1.9-.3 1.7 1.7 0 0 0-1 1.5V21a2 2 0 1 1-4 0v-.2a1.7 1.7 0 0 0-1-1.5 1.7 1.7 0 0 0-1.9.3l-.1.1a2 2 0 1 1-2.8-2.8l.1-.1a1.7 1.7 0 0 0 .3-1.9 1.7 1.7 0 0 0-1.5-1H3a2 2 0 1 1 0-4h.2a1.7 1.7 0 0 0 1.5-1 1.7 1.7 0 0 0-.3-1.9l-.1-.1a2 2 0 1 1 2.8-2.8l.1.1a1.7 1.7 0 0 0 1.9.3h.1a1.7 1.7 0 0 0 1-1.5V3a2 2 0 1 1 4 0v.2a1.7 1.7 0 0 0 1 1.5h.1a1.7 1.7 0 0 0 1.9-.3l.1-.1a2 2 0 1 1 2.8 2.8l-.1.1a1.7 1.7 0 0 0-.3 1.9v.1a1.7 1.7 0 0 0 1.5 1H21a2 2 0 1 1 0 4h-.2a1.7 1.7 0 0 0-1.5 1Z"
+                  />
+                </svg>
```
Comment (mbunwe-victor) [inline src/pages/PresentationActivityPage.tsx]: I noticed that the Activity page header also contains the settings icon, which I suppose was a mislead from the Figma design. Could you remove it?

**Review by mbunwe-victor — CHANGES_REQUESTED** [review]
Hello @Vitalisn4, I have a few comments I would like you to check. Also, the screens you shared earlier looked good, and now I am waiting for the delete card as we spoke about, if you can share a screenshot.

**Comment by mbunwe-victor** [comment]
image


The text is so centralised, and compared with the positioning of text in the Figma design, it doesn't align. Could you check that too?

**Comment by Vitalisn4** [comment]
Here is the confirm deletion
image

**Comment by Vitalisn4** [comment]
> image
> 
> The text is so centralised, and compared with the positioning of text in the Figma design, it doesn't align. Could you check that too?

Addressed

File: src/components/Header.tsx
Comment (Vitalisn4) [reply src/components/Header.tsx:1]: > I honestly don't know why we have all these changes here

That's for the Shared Claims page to mirror our existing wallet in the word size and card thickness

Comment (mbunwe-victor) [reply src/components/Header.tsx:1]: Oh, I see, thanks. I'd prefer we stay consistent with the existing wallet pattern link in `CredentialTypesPage` and `CredentialTypeDetailsPage,` where we made use of a `sub-header.` Rather than adding a `compact` mode to the shared Header. WDYT

Comment (Vitalisn4) [reply src/components/Header.tsx:1]: > Oh, I see, thanks. I'd prefer we stay consistent with the existing wallet pattern link in `CredentialTypesPage` and `CredentialTypeDetailsPage,` where we made use of a `sub-header.` Rather than adding a `compact` mode to the shared Header. WDYT

That makes sense, meaning the header for that Share Claims will have to be left as big as it was before trying to reduce right?

Comment (mbunwe-victor) [reply src/components/Header.tsx:1]: No, if it should follow the flow in the files I mentioned, I suppose the only thing that needs to change is the title, and everything else stays the same.

Comment (Vitalisn4) [reply src/components/Header.tsx:1]: > No, if it should follow the flow in the files I mentioned, I suppose the only thing that needs to change is the title, and everything else stays the same.

Great, working on that

**Comment by Vitalisn4** [comment]
Hello @mbunwe-victor @Ngha-Boris , can you check again now, I have implemented the reviews
